### PR TITLE
add react-tree-graph types

### PR DESCRIPTION
**Pull request by PCOffline**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-tree-graph/index.d.ts
1. types/react-tree-graph/react-tree-graph-tests.tsx

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a81f11281201537387b064c313af9346d0e3867a
</p>

**Comment by PCOffline** [comment]
@jpb12 can you make sure the types correspond to what you'd expect?

**Comment by jpb12** [comment]
The `Data` interface has the following optional parameters:
`keyProp`, `labelProp`, `nodeShape`, `nodeProps`, `gProps`, `pathProps` & `textProps`
See the [overriding props](https://github.com/jpb12/react-tree-graph?tab=readme-ov-file#overriding-props) section of the readme for details.

I'm not sure how you would show this in typescript but every time a property is of type `HTMLProps<SVGSomething>`, we pass additional properties into any event handlers. See [events](https://github.com/jpb12/react-tree-graph?tab=readme-ov-file#events).

**Comment by PCOffline** [comment]
Oh shit, I missed that! I have an idea how to implement it, I'll work on it in a couple of days. Thanks for the review :)

File: types/react-tree-graph/react-tree-graph-tests.js
Comment (sheetalkamat) [inline types/react-tree-graph/react-tree-graph-tests.js:1]: Delete this empty file

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-tree-graph/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against acb98f4d397c5deee40264a425052938435db9e2
</p>

**Comment by PCOffline** [comment]
@jpb12 added the overriding fields and event listeners, can you just verify it works as intended?

File: types/react-tree-graph/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    keyProp?: string;
+    labelProp?: string;
+    shape?: "circle" | "image" | "polygon" | "rect";
+    nodeProps?:
+        | AddNodeIdToElementEvents<HTMLProps<SVGCircleElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGRectElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGImageElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGPolygonElement>>;
+    gProps?: AddNodeIdToElementEvents<HTMLProps<SVGGElement>>;
+    pathProps?: AddNodeIdToElementEvents<HTMLProps<SVGPathElement>>;
```
Comment (jpb12) [inline types/react-tree-graph/index.d.ts:21]: `pathProps` has a different method signature to the others, as it adds both the source and target IDs to the event. Something like:

```
type EventListenerWithPathIds<T> = T extends ((this: infer ListenerThis, ev: infer Event) => infer ListenerReturnType)
     ? (this: ListenerThis, ev: Event, sourceNodeId: string, targetNodeId: string) => ListenerReturnType
     : T;
```

**Review by jpb12 — CHANGES_REQUESTED** [review]

File: types/react-tree-graph/index.d.ts (hunk 1 above)
Comment (PCOffline) [reply types/react-tree-graph/index.d.ts:21]: Yeah noticed it the second I pushed, I'll upload a new commit in a sec

**Review by jpb12 — APPROVED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by PCOffline** [comment]
Ready to merge

**Pull request merged**